### PR TITLE
Document System.IO.Enumeration.FileSystemEnumerable

### PR DESCRIPTION
**Pull request by carlossanlop**
Missing constructor documentation.

### CODE REVIEW

File: xml/System.IO.Enumeration/FileSystemEnumerable`1.xml [hunk 1]
```diff
@@ -54,10 +54,10 @@
         <Parameter Name="options" Type="System.IO.EnumerationOptions" />
       </Parameters>
       <Docs>
-        <param name="directory">To be added.</param>
-        <param name="transform">To be added.</param>
-        <param name="options">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="directory">The path of the directory where the enumeration will be performed.</param>
```
Comment (mairaw) [inline xml/System.IO.Enumeration/FileSystemEnumerable`1.xml]: do we need to say whether this is a full or relative path?

File: xml/System.IO.Enumeration/FileSystemEnumerable`1.xml [hunk 2]
```diff
@@ -54,10 +54,10 @@
...
+        <param name="transform">A delegate method for transforming raw find data into a result.</param>
+        <param name="options">An object describing the enumeration options.</param>
+        <summary>Initializes a new instance of the <see cref="T:System.IO.Enumeration.FileSystemEnumerable`1" /> class with specific search and filtering options.</summary>
         <remarks>To be added.</remarks>
```
Comment (mairaw) [inline xml/System.IO.Enumeration/FileSystemEnumerable`1.xml]: do you want to add basic exceptions such as ArgumentNullException when directory or transform is null?

Also, it seems that if you pass null to options, its default value will be EnumerationOptions.Default. Worth mentioning?

File: xml/System.IO.Enumeration/FileSystemEnumerable`1.xml (hunk 1 above)
Comment (carlossanlop) [reply xml/System.IO.Enumeration/FileSystemEnumerable`1.xml]: It doesn't matter. I followed the code path this constructor takes, and we eventually try to normalize the path / get the full path here: https://source.dot.net/#System.IO.FileSystem/System/IO/Enumeration/FileSystemEnumerator.cs,38

File: xml/System.IO.Enumeration/FileSystemEnumerable`1.xml (hunk 2 above)
Comment (carlossanlop) [reply xml/System.IO.Enumeration/FileSystemEnumerable`1.xml]: Yes, good idea:

```suggestion
         <remarks>
           <format type="text/markdown"><![CDATA[

## Remarks

If `options` is passed as `null`, it defaults to <xref:System.IO.EnumerationOptions.Default>.

           ]]></format>
         </remarks>
         <exception cref="T:System.ArgumentNullException"><paramref name="directory" /> or <paramref name="transform" /> is <see langword="null" />.</exception>
         <exception cref="T:System.ArgumentException">The <paramref name="directory" /> path is empty.

-or-

<paramref name="directory" /> contains a null character "\0".</exception>
```

**Pull request merged**